docs/StorageManagement.md

```diff
@@ -0,0 +1,281 @@
+// import registrator
+import "registrator"
+// import agents
+import "s3Agent"
+
+... within AllocateStorage
+// get agent by type
+agent = registrator.GetAgent(type)
+// invoke the agent
+agent.AllocateStorage...
+```
+
+## How to support a new connection type
+
+### StorageManager
+
+- Implement `AllocateStorage`/`DeleteStorage` for the new type and register it. (Depends on the definition of the appropriate schema by Fybrik.)
+
+- Create a new docker image of StorageManager
+
+### Fybrik core
+
+- Add a new connection schema and compile `taxonomy.json`.
+
+- Ensure existence of modules that are able to write/copy to this connection. Update the capabilities in module yamls accordingly.
+
+- Prepare FybrikStorageAccount resources with the shared storage information.
+
+- Update infrastructure attributes related to the storage accounts, e.g., cost.
+
+- Optionally update IT config policies to specify when the new storage can/should be selected
+
+- Optionally extend catalog-connector to support the new connection.
+
+- Deploy new/modified yamls and re-install Fybrik release using StorageManager image and the new taxonomy schema. No change to fybrik_crd is required.
+
+
+## Fybrik deployment configuration
+
+In [values.yaml](https://github.com/fybrik/fybrik/blob/master/charts/fybrik/values.yaml) add a section `storageManager` with `image` of StorageManager. Modify manager deployment to bring up a new container running in the manager pod.
+
+
+## Changes to Optimizer and storage type selection
+
+Currently, the only available storage type is S3. It has been hard-coded inside manager as the default connection type used in the write flow of a new dataset. Now, the following changes are required (both to optimizer and the manager naive algorithm):
+
+- add the constraint of storage type/category matching the module protocol
+
+- do not specify the desired connection type and determine it later from the selected storage type
+
+
+## To consider in the future:
+
+- Changes to FybrikApplication (add user requirements for storage type, connection details such as bucket name, etc. to the dataset entry)
+
+- Use information about the amount of storage available and amount of data to be written/copied to influence storage selection.
+
+- Extend IT config policies with options for storage management
+
+## Development plan 
+
+### Phase1
+
+- Provide connection taxonomy layer for s3, db2, kafka, arrow-flight, what else?
+
+- Changes to FybrikStorageAccount CR
+
+- Implement StorageManager with the defined API for s3 using minio sdk.
+
+- Remove dependency on datashim
+
+- Update documentation accordingly
+
+- Changes to Airbyte module to adapt the suggested taxonomy
+
+
+### Phase2
+
+- Lift the requirement for the default S3 storage, add constraints to the optimizer. Change the non-CSP algorithm as well.
+
+- Support more storage types: db2, kafka, google sheets, mySQL
+
+### Phase3
+
+- IT config policies for configuration options
+
+### Phase4
+
```

Comment:
why do you separate support connection types and storage types? Should they be implemented together ?